doc/cprover-manual/contracts-decreases.md

```diff
@@ -1,3 +1,179 @@
-# Decreases Clause
+# Decreases Clauses
 
-TODO: Document `__CPROVER_decreases`
+A _decreases_ clause specifies a measure that must strictly decrease at every iteration of a loop.
+By demonstrating that the measure
+1. is bounded from below, and
+2. strictly decreases at each iteration,
+we can prove termination of loops.
+This is because the measure must eventually hit the lower bound
+at which point the loop must terminate,
+since the measure cannot strictly decrease further.
+
+### Syntax
+
+A one-dimensional (1D) decreases clause for a loop is an arithmetic expression `e`
+over the variables visible at the same scope as the loop,
+specified as `__CPROVER_decreases(e)`.
```

Comment:
It doesn't have to be only arithmetic expressions, right? What about `__CPROVER_decreases(i)`?

Reply:
Numeric variables by themselves are valid arithmetic expressions too -- [any expression whose return type is numeric is an arithmetic expression](https://www.cis.upenn.edu/~matuszek/General/JavaSyntax/arithmetic-expressions.html). `5`, `x` (for numeric `x`), `x+5`, `x <= 0 ? -x : x` are all valid arithmetic expressions.